spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java

```diff
@@ -250,6 +250,20 @@ private void deployStream(StreamDefinition stream, Map<String, String> streamDep
 			Resource appResource = registration.getResource();
 			Resource metadataResource = registration.getMetadataResource();
 
+			// add properties needed for metrics system
+			appDeployTimeProperties.put("spring.cloud.dataflow.stream.name", currentApp.getStreamName());
+			appDeployTimeProperties.put("spring.cloud.dataflow.stream.app.label", currentApp.getName());
+			appDeployTimeProperties.put("spring.cloud.dataflow.stream.app.type", type.toString());
+			StringBuilder sb = new StringBuilder()
+					.append(currentApp.getStreamName()).append(".")
+					.append(currentApp.getName()).append(".")
+					.append("${spring.cloud.application.guid}");
+			appDeployTimeProperties.put("spring.cloud.stream.metrics.key", sb.toString());
```

Comment:
All the `spring.cloud.stream.metrics` properties' keys could probably be defined in `BindingPropertyKeys` Class. Perhaps, we could rename the Class `BindingPropertyKeys` to `SpringCloudStreamPropertyKeys` which would also lead a way to create `SpringCloudDataFlowPropertyKeys` for other property keys that start with prefix `spring.cloud.dataflow`.